src/compiler/checker.ts

```diff
@@ -10547,62 +10547,102 @@ module ts {
             return false;
         }
 
-        function getSymbolsInScope(location: Node, meaning: SymbolFlags): Symbol[] {
+        function getSymbolsInScope(location: Node, meaning: SymbolFlags, predicate?: (symbol: Symbol) => boolean): Symbol[] {

```

Comment:
Looking at the numbers, it appears to save a few ms (4-5) for global completion lists.  I think that does make it worth it.  My concern is arond a project with several large d.ts files (like WinRT, etc.)  With all the tens of thousands of symbols, it would be nice to not have to go get *all* of them again since we only want one.

But, i'll defer to you here?  I think it would be fine to remove the predicate.